napalm_yang/mappings/eos/parsers/config/openconfig-system/system.yaml

```diff
@@ -0,0 +1,367 @@
+    execute:
+        - method: cli
+          kwargs:
+              commands: ["show running-config all"]
+system:
+    _process: unnecessary
+    aaa:
+        _process: unnecessary
+        accounting:
+            _process: not_implemented
+            config:
+                _process: not_implemented
+                accounting-method:
+                    _process: not_implemented
+            events:
+                _process: not_implemented
+                event:
+                    _process: not_implemented
+                    config:
+                        _process: not_implemented
+                        event-type:
+                            _process: not_implemented
+                        record:
+                            _process: not_implemented
+                    event-type:
+                        _process: not_implemented
+                    state:
+                        _process: not_implemented
+            state:
+                _process: not_implemented
+        authentication:
+            _process: unnecessary
+            admin-user:
+                _process: not_implemented
+                config:
+                    _process: not_implemented
+                    admin-password:
+                        _process: not_implemented
+                    admin-password-encrypted:
+                        _process: not_implemented
+                state:
+                    _process: not_implemented
+            config:
+                _process: not_implemented
+                authentication-method:
+                    _process: not_implemented
+            state:
+                _process: not_implemented
+            users:
+                _process: unnecessary
+                user:
+                    _process:
+                        - path: "username"
+                          from: root_system.0
+                    config:
+                        _process: unnecessary
+                        password:
+                            _process: not_implemented
+                        password-encrypted:
+                            # Since we are using the show run all command
+                            # the username commands will always start with privilege
+                            _process:
+                                - path: "privilege"
+                                  regexp: "\\d+ role \\S+ secret (5|sha512) (?P<value>\\S+)$"
+                                - path: "privilege"
+                                  regexp: "\\d+ secret (5|sha512) (?P<value>\\S+)$"
+                        #
+                        # In EOS there is privilege and roles which are similar
+                        # here we try to grab both if they are set or just one
+                        # luckily the ordering is consistent regardless how it
+                        # was entered(in eos at least), privilege then role
+                        #
+                        role:
```

Comment:
I was wondering if we should do something like `$privilege:$role`:

```
_process:
    - path: "privilege"
      regexp: "(?P<value>\\d+) secret"
      value: "{{ value }}:network-operator"
    - path: "privilege"
      regexp: "(?P<value>(?P<privilege>\\d+) role (?P<role>\\S+)) secret" 
      value: "{{ privilege }}:{{ role }}"
```

I just checked and looks like "network-operator" is the default role so I guess it would be better to be consistent and always include it. Thoughts?

Reply:
I like that idea.  Unfortunately I can't get this to work as expected with the mock data I have.  Here is what I have now

```
role:
     _process:
             - path: "privilege"
                regexp: "(?P<value>\\d+) nopassword"
                value: "{{value}}:network-operator"
             - path: "privilege"
                regexp: "(?P<value>\\d+) secret \\S+"
                 value: "{{value}}:network-operator"
              - path: "privilege"
                 regexp: "(?P<value>(?P<privilege>\\d+) role (?P<role>\\S+)) secret \\S+"
                  value: "{{ privilege }}:{{ role }}"
```

I get this traceback
```
    yang.parse_config(device=d)
  File "/Users/tim/napalm-yang/napalm_yang/base.py", line 241, in parse_config
    parser.parse()
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 87, in parse
    self._parse(self._yang_name, self.model, self.mapping[self._yang_name])
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 93, in _parse
    self._parse_container(attribute, model, mapping)
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 131, in _parse_container
    self._parse(k, v, mapping[v._yang_name])
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 93, in _parse
    self._parse_container(attribute, model, mapping)
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 131, in _parse_container
    self._parse(k, v, mapping[v._yang_name])
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 93, in _parse
    self._parse_container(attribute, model, mapping)
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 131, in _parse_container
    self._parse(k, v, mapping[v._yang_name])
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 93, in _parse
    self._parse_container(attribute, model, mapping)
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 131, in _parse_container
    self._parse(k, v, mapping[v._yang_name])
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 95, in _parse
    self._parse_list(attribute, model, mapping)
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 173, in _parse_list
    self._parse(key, obj, element_mapping)
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 93, in _parse
    self._parse_container(attribute, model, mapping)
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 131, in _parse_container
    self._parse(k, v, mapping[v._yang_name])
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 93, in _parse
    self._parse_container(attribute, model, mapping)
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 131, in _parse_container
    self._parse(k, v, mapping[v._yang_name])
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 97, in _parse
    self._parse_leaf(attribute, model, mapping)
  File "/Users/tim/napalm-yang/napalm_yang/parser.py", line 185, in _parse_leaf
    value = self.parser.parse_leaf(attribute, mapping["_process"], self.bookmarks)
  File "/Users/tim/napalm-yang/napalm_yang/parsers/base.py", line 151, in parse_leaf
    result = self._parse_leaf_default(attribute, m, data)
  File "/Users/tim/napalm-yang/napalm_yang/parsers/text_tree.py", line 113, in _parse_leaf_default
    return super()._parse_leaf_default(attribute, mapping, data)
  File "/Users/tim/napalm-yang/napalm_yang/parsers/jsonp.py", line 101, in _parse_leaf_default
    d = self._parse_post_process_filter(mapping["value"], value=d)
  File "/Users/tim/napalm-yang/napalm_yang/parsers/base.py", line 183, in _parse_post_process_filter
    return helpers.template(post_process_filter, extra_vars=self.extra_vars, **kwargs)
  File "/Users/tim/napalm-yang/napalm_yang/helpers.py", line 137, in template
    return template.render(**kwargs)
  File "/usr/local/lib/python2.7/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/lib/python2.7/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "<template>", line 1, in top-level template code
jinja2.exceptions.UndefinedError: 'privilege' is undefined
```
I tired to do `extra_vars.privilege`, since that how it is done with the address prefix in the ipv4.yaml EOS mapping, but it fails as well but because with this error
`jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'privilege'`

Right before the template.render command in helpers I launched a debugger.  It looks like the 
```
string = {{ privilege }}:{{ role }}
kwargs = {'user_key': 'test3', 'parent_key': 'test3', 'value': '15 role network-operator secret sha512 $6$Vd6.7k2FybfsTKKp$S.AHfdwicaWEoA41sPd6ZXOOdruJMrKJh70WNfiX/eZKH1oYBtFz9VbrPlYNDkhM/pi54gcYKH2hviy/xrUav.', 'extra_vars': {'user': {}, 'parent': {}}}
```

The other thing we'll need to do is check for the following line to set the default role
`aaa authorization policy local default-role network-admin` and set the default role accordingly.

Reply:
No worries, will try to find some time tomorrow and give it a try myself. Might require some massaging in the core.